projects/dex-ui/src/wells/useWellSwaps.tsx

```diff
@@ -0,0 +1,69 @@
+import { useEffect, useState } from "react";
+import request from "graphql-request";
+import { useQuery } from "@tanstack/react-query";
+import { GetWellSwapsDocument } from "src/generated/graphql";
+
+import { TokenValue, ERC20Token } from "@beanstalk/sdk";
+import { useNetwork } from "wagmi";
+import { fetchFromSubgraphRequest } from "./subgraphFetch";
+import { useGetWellSwapsQuery } from "src/generated/graph";
+
+export type WellSwap = {
+  hash: string;
+  label: string;
+  account: string;
+  amountIn: TokenValue;
+  tokenIn: ERC20Token;
+  amountOut: TokenValue;
+  tokenOut: ERC20Token;
+  timestamp: string;
+};
+
+const useWellSwaps = (wellId: string) => {
+  const { chain } = useNetwork();
+
+  const {
+    isLoading: loading,
+    error,
+    data
+  } = useQuery(
+    ["wellswaps"],
```

Comment:
we should double check this cache key. it will be overwritten (or wrongly re-used) when browsing to another well